test/Analysis/test-allocation.mlir

```diff
@@ -9,6 +9,8 @@
 #A_DOT = #triton_gpu.dot_op<{opIdx = 0, parent = #C}>
 #B_DOT = #triton_gpu.dot_op<{opIdx = 1, parent = #C}>
 
+module attributes {"triton_gpu.num-warps" = 4 : i32} {
```

Comment:
We need attribute `triton_gpu.num-warps` to calculate shared memory usage for fast reduce